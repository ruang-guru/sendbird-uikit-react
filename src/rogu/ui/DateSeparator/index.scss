@import "../../../styles/variables";

.rogu-date-separator {
  width: 100%;
  text-align: center;
  margin-bottom: 1rem;
  position: sticky;
  top: 1rem;
  bottom: 1rem;

  .rogu-date-separator__content {
    padding: 4px 8px;
    border-radius: 4px;
    text-transform: uppercase;

    @include themed() {
      color: t(on-bg-1);
      background-color: t(bg-1);
    }
  }
}
