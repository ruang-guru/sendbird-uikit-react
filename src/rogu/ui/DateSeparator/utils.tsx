import { differenceInCalendarDays, format, getDay } from "date-fns";

import { getDayString, DayStringType } from "../../utils";

export type DateSeparatorStringsType = {
  today: string;
  yesterday: string;
  days: DayStringType;
};

export const getDateSeparatorDifference = (
  createdAt: Date | number,
  strings: DateSeparatorStringsType
): string => {
  const diffWithToday = differenceInCalendarDays(new Date(), createdAt);

  if (diffWithToday === 0) {
    return strings.today;
  } else if (diffWithToday === 1) {
    return strings.yesterday;
  } else if (diffWithToday <= 7) {
    return getDayString(getDay(createdAt), strings.days);
  } else {
    return format(createdAt, "dd/MM/yyyy");
  }
};

export default getDateSeparatorDifference;
