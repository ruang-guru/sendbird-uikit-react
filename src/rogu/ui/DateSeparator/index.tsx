import React, { useContext } from "react";

import "./index.scss";
import Label, { LabelTypography } from "../Label";

import { getClassName } from "../../../utils";
import { getDateSeparatorDifference } from "./utils";
import { LocalizationContext } from "../../../lib/LocalizationContext";

export type DateSeparatorProps = {
  className?: string;
  createdAt: Date | number;
};

export default function DateSeparator({
  className,
  createdAt,
}: DateSeparatorProps) {
  const { stringSet } = useContext(LocalizationContext);

  return (
    <div className={getClassName([className, "rogu-date-separator"])}>
      <Label
        className="rogu-date-separator__content"
        type={LabelTypography.CAPTION_1}
      >
        {getDateSeparatorDifference(createdAt, {
          today: stringSet.LABEL__DATE_TODAY,
          yesterday: stringSet.LABEL__DATE_YESTERDAY,
          days: {
            0: stringSet.LABEL__DAY_SUNDAY,
            1: stringSet.LABEL__DAY_MONDAY,
            2: stringSet.LABEL__DAY_TUESDAY,
            3: stringSet.LABEL__DAY_WEDNESDAY,
            4: stringSet.LABEL__DAY_THURSDAY,
            5: stringSet.LABEL__DAY_FRIDAY,
            6: stringSet.LABEL__DAY_SATURDAY,
          },
        })}
      </Label>
    </div>
  );
}
