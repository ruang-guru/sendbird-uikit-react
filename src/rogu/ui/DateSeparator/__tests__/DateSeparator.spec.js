import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import DateSeparator from '../index';

describe('DateSeparator', () => {
  it('should contain className', function () {
    const className = 'example-classname';
    const component = shallow(
      <DateSeparator className={className} createdAt={new Date()} />
    );

    expect(component.find('.rogu-date-separator').hasClass(className)).toBe(
      true
    );
  });

  it('should do a snapshot test of the DateSeparator DOM', function () {
    const className = 'ÃŸ-classname';
    const component = renderer.create(
      <DateSeparator className={className} createdAt={new Date()} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
