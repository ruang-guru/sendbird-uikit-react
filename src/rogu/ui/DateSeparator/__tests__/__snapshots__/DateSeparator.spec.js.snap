// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateSeparator should do a snapshot test of the DateSeparator DOM 1`] = `
<div
  className="example-classname sendbird-separator"
>
  <div
    className="sendbird-separator__left sendbird-color--onbackground-4--background-color"
  />
  <div
    className="sendbird-separator__text"
  >
    <span
      className="sendbird-label sendbird-label--caption-2 sendbird-label--color-onbackground-2"
    >
      Date Separator
    </span>
  </div>
  <div
    className="sendbird-separator__right sendbird-color--onbackground-4--background-color"
  />
</div>
`;
